src/java.base/share/classes/sun/security/tools/keytool/Main.java

```diff
@@ -1149,7 +1159,8 @@ void doCommands(PrintStream out) throws Exception {
                 throw new Exception(rb.getString(
                         "keyalg.option.missing.error"));
             }
-            doGenKeyPair(alias, dname, keyAlgName, keysize, groupName, sigAlgName);
+            doGenKeyPair(alias, dname, keyAlgName, keysize, groupName, sigAlgName,
+                    signerAlias);
```

Comment:
Maybe we can just keep using the old argument list. `signerAlias` is a global variable and it's visible everywhere.

Reply:
I'd prefer to have it. It is the same as other global arguments being passed to this method.